org-formation/090-systems-manager/ScheduledMaintenance.yaml

```diff
@@ -0,0 +1,127 @@
+    Description: >-
+      space delimited list of key:value, where key is an SSM Parameter Store
+      key and value is the shell envvar name to assign.
+Resources:
+  # Create IAM Role for SSM
+  SSMExecutionRole:
+    Type: "AWS::IAM::Role"
+    Properties:
+      AssumeRolePolicyDocument:
+        Version: '2012-10-17'
+        Statement:
+          - Effect: Allow
+            Principal:
+              Service: "ssm.amazonaws.com"
+            Action: "sts:AssumeRole"
+      Policies:
+        - PolicyName: "SSMExecutionPolicy"
+          PolicyDocument:
+            Version: '2012-10-17'
+            Statement:
+              - Effect: Allow
+                Action:
+                  - "ec2:DescribeInstances"
+                  - "ssm:SendCommand"
+                Resource: "*"
+        - PolicyName: "SSMParameterPolicy"
+          PolicyDocument:
+            Version: '2012-10-17'
+            Statement:
+              - Effect: Allow
+                Action:
+                  - "ssm:GetParameters"
+                Resource:
+                - !Join [ '', ['arn:', !Ref 'AWS::Partition', ':ssm:', !Ref 'AWS::Region', ':', !Ref 'AWS::AccountId', '*'] ]
+  SSMDocument:
+    Type: "AWS::SSM::Document"
+    Properties:
+      DocumentType: "Command"
+      Content:
+        schemaVersion: "2.2"
+        description: "Run a script on EC2 instances"
+        parameters:
+          ScriptUrl:
+            type: String
+            description: "The URL of the script to run"
+          Region:
+            type: String
+            description: The AWS region
+          SsmParameterNameToEnvvarMap:
+            type: String
+            description: >-
+              space delimited list of key:value, where key is an SSM Parameter Store
+              key and value is the shell envvar name to assign.
+        mainSteps:
+        - action: "aws:runShellScript"
+          name: "runShellScript"
+          inputs:
+            runCommand:
+            - SSM_TO_ENV_MAP=( {{ SsmParameterNameToEnvvarMap }} )
+            - for KV in ${SSM_TO_ENV_MAP[@]} ; do
+            -   export ${KV##*:}=$(aws --region {{ Region }} ssm get-parameters --names ${KV%%:*} --with-decryption --query Parameters[0].Value --output text)
+            - done
+            - SCRIPT_PATH=/tmp/script.sh
+            - wget -O $SCRIPT_PATH {{ ScriptUrl }}
+            - chmod +x $SCRIPT_PATH
+            - sudo $SCRIPT_PATH
+
+  # Create Maintenance Window
+  MaintenanceWindow:
+    Type: "AWS::SSM::MaintenanceWindow"
+    Properties:
+      Name: "DailyMaintenanceWindow"
+      Schedule: !Ref Schedule
+      Duration: 1 # Duration of the maintenance window in hours
+      Cutoff: 0
+      AllowUnassociatedTargets: False
+
+  # Create Maintenance Window Target
+  MaintenanceWindowTarget:
+    Type: "AWS::SSM::MaintenanceWindowTarget"
+    Properties:
+      WindowId: !Ref MaintenanceWindow
+      ResourceType: "INSTANCE"
+      Targets:
+        - Key: "InstanceIds"
```

Comment:
i think targeting by tags would be better because it's much more flexible.

Reply:
While targeting by tags works, I want to make sure that instances are scanned by default, not only when they are correctly tagged.  I want to catch the case in which someone provisions an instance, doesn't set it up the way we'd like, and then doesn't patch vulnerabilities.  If we require a specific tag to scan the instance then we could have a 'blind spot.'